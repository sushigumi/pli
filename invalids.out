Testing invalid syntax
Testing tests/invalids/beginMisspell.gt
Parser error at (line 3, column 1):
unexpected "g"
expecting type declaration or "begin"

Testing tests/invalids/call.gt
Parser error at (line 4, column 15):
unexpected ":"
expecting variable, operator, "," or ")"

Testing tests/invalids/call1.gt
proc main ()
    int m;
begin
    call id(id);
end

Testing tests/invalids/funcExtraComma.gt
Parser error at (line 1, column 20):
unexpected ")"
expecting function parameter starting with 'ref' or 'val'

Testing tests/invalids/invalidArray1.gt
Parser error at (line 2, column 11):
unexpected "]"
expecting size or initializer for variable with array type

Testing tests/invalids/invalidArray2.gt
Parser error at (line 2, column 13):
unexpected "]"
expecting ']', size or initializer for array variable

Testing tests/invalids/invalidArray3.gt
Parser error at (line 2, column 15):
unexpected "\n"
expecting ";"

Testing tests/invalids/invalidArray4.gt
Parser error at (line 2, column 11):
unexpected ","
expecting size or initializer for variable with array type

Testing tests/invalids/invalidIf1.gt
Parser error at (line 5, column 9):
unexpected "m"
expecting ".", operator, "=", "!=", "<", "<=", ">", ">=" or "then"

Testing tests/invalids/invalidIf2.gt
Parser error at (line 7, column 4):
unexpected reserved word "end"
expecting statement

Testing tests/invalids/invalidIf3-1.gt
Parser error at (line 4, column 12):
unexpected reserved word "read"
expecting expression

Testing tests/invalids/invalidIf3-2.gt
Parser error at (line 4, column 11):
unexpected reserved word "int"
expecting expression

Testing tests/invalids/invalidIf4-1.gt
Parser error at (line 5, column 11):
unexpected "<"
expecting variable or ":="

Testing tests/invalids/invalidIf4-2.gt
Parser error at (line 5, column 12):
unexpected reserved word "int"
expecting at least 1 statement

Testing tests/invalids/invalidWhile1.gt
Parser error at (line 5, column 9):
unexpected "m"
expecting ".", operator, "=", "!=", "<", "<=", ">", ">=" or "do"

Testing tests/invalids/invalidWhile2.gt
Parser error at (line 6, column 4):
unexpected reserved word "end"
expecting statement

Testing tests/invalids/invalidWhile3-1.gt
Parser error at (line 4, column 13):
unexpected ":"
expecting variable, operator or "do"

Testing tests/invalids/invalidWhile3-2.gt
Parser error at (line 4, column 14):
unexpected reserved word "int"
expecting expression

Testing tests/invalids/invalidWhile4-1.gt
Parser error at (line 5, column 11):
unexpected ">"
expecting variable or ":="

Testing tests/invalids/invalidWrite.gt
Parser error at (line 4, column 13):
unexpected "\n"
expecting operator or ";"

Testing tests/invalids/invalidWriteNewline.gt
Parser error at (line 4, column 16):
unexpected "\n"
expecting "\""

Testing tests/invalids/invalidWriteTab.gt
proc main ()
    int m;
begin
    write "asdf\t";
end

Testing tests/invalids/missingBegin.gt
Parser error at (line 4, column 5):
unexpected "r"
expecting type declaration or "begin"

Testing tests/invalids/missingBrackets.gt
Parser error at (line 2, column 5):
unexpected "i"
expecting "("

Testing tests/invalids/missingEnd.gt
Parser error at (line 6, column 1):
unexpected end of input
expecting statement or "end"

Testing tests/invalids/missingFuncIdent.gt
Parser error at (line 1, column 6):
unexpected "("
expecting function name

Testing tests/invalids/missingMain.gt
proc main ()
    int m;
begin
    read m;
end

proc main ()
    int n;
begin
    read n;
end

Testing tests/invalids/missingRef.gt
Parser error at (line 1, column 9):
unexpected "i"
expecting function parameter starting with 'ref' or 'val' or ")"

Testing tests/invalids/missingSemiDecl.gt
Parser error at (line 3, column 1):
unexpected "b"
expecting variable or ";"

Testing tests/invalids/missingSemiStmt.gt
Parser error at (line 5, column 1):
unexpected "e"
expecting variable or ";"

Testing tests/invalids/readExpr.gt
Parser error at (line 4, column 12):
unexpected "<"
expecting variable or ";"

Testing tests/invalids/readStmt.gt
Parser error at (line 4, column 12):
unexpected ":"
expecting variable or ";"

Testing tests/invalids/tripleEquals.gt
Parser error at (line 4, column 17):
unexpected "o"
ambiguous use of a non associative operator

Testing tests/invalids/twoMains.gt
proc main ()
    int m;
begin
    m := 1;
end

proc main ()
    int m;
begin
    m := 1;
end

Testing tests/invalids/twoPlus.gt
Parser error at (line 4, column 14):
unexpected "+"
expecting constant, "(" or identifier

Testing tests/invalids/twoSemis.gt
Parser error at (line 2, column 11):
unexpected ";"
expecting type declaration or "begin"

Testing tests/invalids/writeId.gt
proc main ()
    int m;
begin
    write m;
end

Testing tests/invalids/writeStmt.gt
Parser error at (line 4, column 13):
unexpected ":"
expecting variable, operator or ";"

