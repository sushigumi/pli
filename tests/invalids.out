Testing invalid syntax
Testing invalids/beginMisspell.gt
Parser error at (line 3, column 1):
unexpected "g"
expecting type declaration or "begin"

Testing invalids/call1.gt
proc main ()
    int m;
begin
    call id(id);
end

Testing invalids/call.gt
Parser error at (line 4, column 15):
unexpected ":"
expecting variable, operator, "," or ")"

Testing invalids/funcExtraComma.gt
Parser error at (line 1, column 20):
unexpected ")"
expecting function parameter starting with 'ref' or 'val'

Testing invalids/invalidArray1.gt
Parser error at (line 2, column 11):
unexpected "]"
expecting size or initializer for variable with array type

Testing invalids/invalidArray2.gt
Parser error at (line 2, column 13):
unexpected "]"
expecting ']', size or initializer for array variable

Testing invalids/invalidArray3.gt
Parser error at (line 3, column 1):
unexpected "b"
expecting ";"

Testing invalids/invalidArray4.gt
Parser error at (line 2, column 11):
unexpected ","
expecting size or initializer for variable with array type

Testing invalids/invalidIf1.gt
Parser error at (line 5, column 9):
unexpected "m"
expecting ".", operator, "=", "!=", "<", "<=", ">", ">=" or "then"

Testing invalids/invalidIf2.gt
Parser error at (line 7, column 4):
unexpected reserved word "end"
expecting statement

Testing invalids/invalidIf3-1.gt
Parser error at (line 4, column 12):
unexpected reserved word "read"
expecting expression

Testing invalids/invalidIf3-2.gt
Parser error at (line 4, column 11):
unexpected reserved word "int"
expecting expression

Testing invalids/invalidIf4-1.gt
Parser error at (line 5, column 11):
unexpected "<"
expecting variable or ":="

Testing invalids/invalidIf4-2.gt
Parser error at (line 5, column 12):
unexpected reserved word "int"
expecting at least 1 statement

Testing invalids/invalidWhile1.gt
Parser error at (line 5, column 9):
unexpected "m"
expecting ".", operator, "=", "!=", "<", "<=", ">", ">=" or "do"

Testing invalids/invalidWhile2.gt
Parser error at (line 6, column 4):
unexpected reserved word "end"
expecting statement

Testing invalids/invalidWhile3-1.gt
Parser error at (line 4, column 13):
unexpected ":"
expecting variable, operator or "do"

Testing invalids/invalidWhile3-2.gt
Parser error at (line 4, column 14):
unexpected reserved word "int"
expecting expression

Testing invalids/invalidWhile4-1.gt
Parser error at (line 5, column 11):
unexpected ">"
expecting variable or ":="

Testing invalids/invalidWrite.gt
Parser error at (line 4, column 13):
unexpected "\n"
expecting operator or ";"

Testing invalids/invalidWriteNewline.gt
Parser error at (line 4, column 16):
unexpected "\n"
expecting "\""

Testing invalids/invalidWriteTab.gt
proc main ()
    int m;
begin
    write "asdf\t";
end

Testing invalids/missingBegin.gt
Parser error at (line 4, column 5):
unexpected "r"
expecting type declaration or "begin"

Testing invalids/missingBrackets.gt
Parser error at (line 2, column 5):
unexpected "i"
expecting "("

Testing invalids/missingEnd.gt
Parser error at (line 6, column 1):
unexpected end of input
expecting statement or "end"

Testing invalids/missingFuncIdent.gt
Parser error at (line 1, column 6):
unexpected "("
expecting function name

Testing invalids/missingMain.gt
proc main ()
    int m;
begin
    read m;
end

proc main ()
    int n;
begin
    read n;
end

Testing invalids/missingRef.gt
Parser error at (line 1, column 9):
unexpected "i"
expecting function parameter starting with 'ref' or 'val' or ")"

Testing invalids/missingSemiDecl.gt
Parser error at (line 3, column 1):
unexpected "b"
expecting variable or ";"

Testing invalids/missingSemiStmt.gt
Parser error at (line 5, column 1):
unexpected "e"
expecting variable or ";"

Testing invalids/readExpr.gt
Parser error at (line 4, column 12):
unexpected "<"
expecting variable or ";"

Testing invalids/readStmt.gt
Parser error at (line 4, column 12):
unexpected ":"
expecting variable or ";"

Testing invalids/tripleEquals.gt
Parser error at (line 4, column 17):
unexpected "o"
ambiguous use of a non associative operator

Testing invalids/twoMains.gt
proc main ()
    int m;
begin
    m := 1;
end

proc main ()
    int m;
begin
    m := 1;
end

Testing invalids/twoPlus.gt
Parser error at (line 4, column 14):
unexpected "+"
expecting constant, "(" or identifier

Testing invalids/twoSemis.gt
Parser error at (line 2, column 11):
unexpected ";"
expecting type declaration or "begin"

Testing invalids/writeId.gt
proc main ()
    int m;
begin
    write m;
end

Testing invalids/writeStmt.gt
Parser error at (line 4, column 13):
unexpected ":"
expecting variable, operator or ";"

